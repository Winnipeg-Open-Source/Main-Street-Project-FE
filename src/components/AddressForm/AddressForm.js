import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'pcln-design-system';
import Input from 'components/Input';
import ProvinceSelect from 'components/ProvinceSelect';

function AddressForm ({
    address,
    city,
    province,
    postalCode,
    onAddressChange,
    onCityChange,
    onProvinceChange,
    onPostalCodeChange,
}) {
    return (
        <>
            <Input
                id='address'
                name='address'
                placeholder='Street Address'
                value={address}
                onChange={onAddressChange}
            />
            <Flex width={1}>
                <Input
                    id='city'
                    name='city'
                    placeholder='City'
                    value={city}
                    onChange={onCityChange}
                    mr={2}
                />
                <ProvinceSelect value={province} onChange={onProvinceChange} />
            </Flex>
            <Input
                id='postal-code'
                name='postalCode'
                placeholder='Postal Code'
                value={postalCode}
                onChange={onPostalCodeChange}
            />
        </>
    );
}

AddressForm.displayName = 'AddressForm';

AddressForm.propTypes = {
    address: PropTypes.string,
    city: PropTypes.string,
    province: PropTypes.string,
    postalCode: PropTypes.string,
    onAddressChange: PropTypes.func,
    onCityChange: PropTypes.func,
    onProvinceChange: PropTypes.func,
    onPostalCodeChange: PropTypes.func,
};

export default memo(AddressForm);