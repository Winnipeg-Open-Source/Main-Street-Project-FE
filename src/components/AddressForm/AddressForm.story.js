import React from 'react';
import AddressForm from './AddressForm';
import List from 'components/List';

export const Default = () => (
    <List>
        <AddressForm />
    </List>
);

export default {
    title: 'AddressForm',
    component: AddressForm,
};
