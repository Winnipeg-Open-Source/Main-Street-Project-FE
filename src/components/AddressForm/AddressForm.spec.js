import React from 'react';
import renderWithTheme from 'tests/helpers/renderWithTheme';
import AddressForm from './AddressForm';

const props = {
    address: '123 Test Street',
    city: 'Winnipeg',
    province: 'MB',
    postalCode: 'ABC123',
};

describe ('AddressForm', () => {
    it ('renders without crashing', () => {
        const { asFragment } = renderWithTheme(<AddressForm />);

        expect(asFragment()).toMatchSnapshot();
    });

    it ('renders properly', () => {
        const { getByPlaceholderText, getByTestId } = renderWithTheme(<AddressForm {...props} />);

        expect(getByPlaceholderText('Street Address').value).toBe('123 Test Street');
        expect(getByPlaceholderText('City').value).toBe('Winnipeg');
        expect(getByPlaceholderText('Postal Code').value).toBe('ABC123');
        expect(getByTestId('province-select').value).toBe('MB');
    });
});
