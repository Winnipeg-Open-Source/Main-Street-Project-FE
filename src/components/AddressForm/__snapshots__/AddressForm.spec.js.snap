// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="sc-pbJYR hNzbLI sc-pkfPC fCxHK"
  >
    <input
      class="sc-pTWqp sc-psDhf bvmjWf"
      font-size="2,,1"
      id="address"
      name="address"
      placeholder="Street Address"
      value=""
    />
    <div
      class="sc-AxjAm sc-AxiKw ivVODM"
      width="1"
    >
      <input
        class="sc-pTWqp sc-psDhf UrUUF"
        font-size="2,,1"
        id="city"
        name="city"
        placeholder="City"
        value=""
      />
      <div
        class="sc-AxjAm sc-AxiKw iSVrPI"
        width="1"
      >
        <select
          class="sc-pZCuu kQVKTB sc-oTaid sc-qQxFs glncUx"
          data-testid="province-select"
          font-size="2,,1"
          id="province"
          name="province"
        >
          <option
            value="AB"
          >
            Alberta
          </option>
          <option
            value="BC"
          >
            British Columbia
          </option>
          <option
            value="MB"
          >
            Manitoba
          </option>
          <option
            value="NB"
          >
            New Brunswick
          </option>
          <option
            value="NL"
          >
            Newfoundland and Labrador
          </option>
          <option
            value="NS"
          >
            Nova Scotia
          </option>
          <option
            value="ON"
          >
            Ontario
          </option>
          <option
            value="PE"
          >
            Prince Edward Island
          </option>
          <option
            value="QC"
          >
            Quebec
          </option>
          <option
            value="SK"
          >
            Saskatchewan
          </option>
          <option
            value="NT"
          >
            Northwest Territories
          </option>
          <option
            value="NU"
          >
            Nunavut
          </option>
          <option
            value="YT"
          >
            Yukon
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="sc-Axmtr gXMRoh sc-fzqPZZ sc-pRFHb JnCiE"
          color="text.light"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
          />
        </svg>
      </div>
    </div>
    <input
      class="sc-pTWqp sc-psDhf bvmjWf"
      font-size="2,,1"
      id="postal-code"
      name="postalCode"
      placeholder="Postal Code"
      value=""
    />
  </div>
</DocumentFragment>
`;
